src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -0,0 +1,435 @@
+                ReadOnlySpan<char> label = pemData[labelRange];
+
+                if (!IsValidLabel(label))
+                {
+                    continue;
+                }
+
+                int postebLength = Posteb.Length + label.Length + Ending.Length;
+                Span<char> postebBuffer = postebLength > 256 ? new char[postebLength] : postebShortBuffer;
+                Posteb.AsSpan().CopyTo(postebBuffer);
+                label.CopyTo(postebBuffer[Posteb.Length..]);
+                Ending.AsSpan().CopyTo(postebBuffer[(Posteb.Length + label.Length)..]);
+                ReadOnlySpan<char> posteb = postebBuffer[..postebLength];
+
+                Range postebLineRange = lineRange;
+                int postebLinePosition = preebLinePosition;
+
+                // in lax decoding a posteb may appear on the same line as the preeb.
+                // start on the current line. We do not need to check that this posteb
+                // comes after the preeb because the preeb's prior content has already
+                // been validated to be whitespace.
+                do
+                {
+                    ReadOnlySpan<char> postebLine = pemData[postebLineRange];
+                    int postebIndex = postebLine.IndexOf(posteb);
+
+                    if (postebIndex == -1)
+                    {
+                        continue;
+                    }
+
+                    (int postebOffset, _) = postebLineRange.GetOffsetAndLength(pemData.Length);
+                    int postebEndIndex = postebOffset + postebIndex + posteb.Length;
+                    Range location = preebStartIndex..postebEndIndex;
+                    Range content = (endLabelIndex + Ending.Length)..(postebOffset + postebIndex);
+
+                    if (!postebLine[(postebIndex + posteb.Length)..].IsWhiteSpace())
+                    {
+                        break;
+                    }
+
+                    if (IsValidBase64(pemData, content, out Range base64range, out int decodedBase64Size))
+                    {
+                        fields = new PemFields(labelRange, base64range, location, decodedBase64Size);
+                        return true;
+                    }
+                    break;
+                }
+                while (TryReadNextLine(pemData, ref postebLinePosition, out postebLineRange));
+            }
+
+            fields = default;
+            return false;
+        }
+
+        private static bool IsValidLabel(ReadOnlySpan<char> data)
+        {
+            static bool IsLabelChar(char c) => c >= 0x21 && c <= 0x7e && c != '-';
+
+            if (data.Length == 0)
+                return true;
+
+            // First character of label must be a labelchar
+            if (!IsLabelChar(data[0]))
+                return false;
+
+            for (int index = 1; index < data.Length; index++)
+            {
+                char c = data[index];
+                if (!IsLabelChar(c) && c != ' ' && c != '-')
```

Comment:
Redundant `c != '-'` check?

Reply:
I don't think so. `IsLabelChar` says "Is this a valid character, except for "-"? The first character in allowed to be anything except a space or hyphen. The other characters may be any of those characters, including the hyphen or space. So it's a double negation. I'll try and make this easier to follow.